usr/share/rear/layout/save/default/950_verify_disklayout_file.sh

```diff
@@ -47,21 +47,39 @@ while read keyword disk_dev disk_size parted_mklabel junk ; do
     # as above layout/prepare/GNU/Linux/100_include_partition_code.sh is the most important one
     # so that it is used here as reference to decide whether or not the entries are correct:
     partitions=()
-    while read keyword dummy part_size part_start part_name part_flags part_dev junk ; do
-        test -b "$part_dev" || broken_part_errors=( "${broken_part_errors[@]}" "$part_dev is not a block device" )
-        is_positive_integer $part_size || broken_part_errors=( "${broken_part_errors[@]}" "$part_dev size $part_size is not a positive integer" )
-        is_nonnegative_integer $part_start || broken_part_errors=( "${broken_part_errors[@]}" "$part_dev start $part_start is not a nonnegative integer" )
-        partitions=( "${partitions[@]}" "$part_dev" )
-        # Using the parted_mklabel fallback behaviour in create_partitions() in prepare/GNU/Linux/100_include_partition_code.sh
-        # only when there is no parted_mklabel value, but when there is a parted_mklabel value use it as is:
-        if ! test "$parted_mklabel" ; then
-            case $part_name in
-                (primary|extended|logical)
-                    parted_mklabel="msdos"
-                    ;;
-            esac
-        fi
-    done < <( grep "^part $disk_dev " "$DISKLAYOUT_FILE" )
+   
+    # cf. https://github.com/rear/rear/issues/2137
+    # s390 - partitions do not list the boot flag
+    # flag is removed from while read ... - this is true for sles also
+    # example
+    # /dev/dasda:24.2GB:dasd:512:4096:dasd:IBM S390 DASD drive:;
+    # 1:98.3kB:524MB:524MB:xfs::;
+    # 2:524MB:24.2GB:23.6GB:::lvm;
+    if [ "$ARCH" == "Linux-s390x" ] || [ "$ARCH" == "Linux-s390" ] ; then
```

Comment:
To simplify such tests perhaps it makes sense
to enhance in usr/share/rear/conf/default.conf
the code part that sets ARCH and REAL_ARCH
from the MACHINE and REAL_MACHINE values
so that on IBM Z a common `MACHINE=s390` is set
that results a common ARCH="Linux-s390" so that on IBM Z
such tests could be simplified to something like
```
if test "$ARCH" = "Linux-s390" ; then
```

Reply:
done.
also simplified prep/GNU/Linux/300_include_grub_tools.sh
should 305_include_s390_tools.sh have an exclusion _(such as if arch is no s390...)_ for running for non-s390 or is there a better way?